jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1769,16 +1789,32 @@ public boolean equalsIgnoreCase(String string) {
 			char charAtO1Last = s1.charAtInternal(s1len - 1, s1Value);
 			char charAtO2Last = s2.charAtInternal(s1len - 1, s2Value);
 
-			if (charAtO1Last != charAtO2Last &&
-					toUpperCase(charAtO1Last) != toUpperCase(charAtO2Last) &&
-					((charAtO1Last <= 255 && charAtO2Last <= 255) || Character.toLowerCase(charAtO1Last) != Character.toLowerCase(charAtO2Last))) {
+			if (charAtO1Last != charAtO2Last
+					&& toUpperCase(charAtO1Last) != toUpperCase(charAtO2Last)
+					&& ((charAtO1Last <= 255 && charAtO2Last <= 255) || Character.toLowerCase(charAtO1Last) != Character.toLowerCase(charAtO2Last))
+					/*[IF Java16]*/
+					&& (!Character.isLowSurrogate(charAtO1Last) || !Character.isLowSurrogate(charAtO2Last))
+					/*[ENDIF] Java16 */
+			) {
 				return false;
 			}
 
-			while (o1 < end - 1) {
+			while (o1 < end) {
```

Comment:
Pls make the change for jdk16 only to avoid changing perf for older versions.

Reply:
Updated so it's for Java 16 only.